src/App.tsx

```diff
@@ -1,34 +1,63 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
+import './App.scss';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { FilterBy } from './types/FilterBy';
 
 export const App: React.FC = () => {
+  const todoByDefault: Todo = {
+    userId: 0,
+    id: 0,
+    title: '',
+    completed: false,
+  };
+
+  const [todo, setTodo] = useState(todoByDefault);
+  const [query, setQuery] = useState('');
+  const [filterBy, setFilterBy] = useState<FilterBy>('all');
+  const [loading, setLoading] = useState(false);
+
   return (
     <>
       <div className="section">
         <div className="container">
           <div className="box">
-            <h1 className="title">Todos:</h1>
+            <h1 className="title">Title</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                query={query}
+                setQuery={setQuery}
+                setFilterBy={setFilterBy}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {loading && <Loader />}
+              <TodoList
+                setTodo={setTodo}
+                query={query}
+                filterBy={filterBy}
+                setLoading={setLoading}
+              />
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {todo.id && (
+        <TodoModal
+          todo={todo}
+          setTodo={setTodo}
+        />
+      )}
```

Comment:
according to previous changes here I would use comparing with the todo existence
`{todo && (
        <TodoModal
          todo={todo}
          setSelectedTodo={setSelectedTodo}
        />
      )}`

Reply:
Yes, just don't start with `{}` as initial todo